extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -167,6 +169,22 @@ public AgroalDataSource createDataSource(String dataSourceName,
         if (dataSourceRuntimeConfig.backgroundValidationInterval.isPresent()) {
             poolConfiguration.validationTimeout(dataSourceRuntimeConfig.backgroundValidationInterval.get());
         }
+        if (dataSourceRuntimeConfig.validationQuerySql.isPresent()) {
+            String validationQuery = dataSourceRuntimeConfig.validationQuerySql.get();
+            poolConfiguration.connectionValidator(new ConnectionValidator() {
+
+                @Override
+                public boolean isValid(Connection connection) {
+                    try (Statement stmt = connection.createStatement()) {
+                        stmt.execute(validationQuery);
+                        return true;
+                    } catch (Exception e) {
+                        log.warn("Connection validation failed", e);
```

Comment:
Wouldn't an `ERROR `log be better?

Reply:
if a validation check of a connection fails, it indicates that the connection needs to be dropped and a new one needs to be provided. As such IMHO this does not necessary indicate an error event, but more of a warning information that this happened. 
But please correct me if I am wrong.

Reply:
No problem :)